x/ecocredit/server/marketplace/utils.go

```diff
@@ -159,6 +159,17 @@ func (k Keeper) fillOrder(ctx context.Context, sellOrder *api.SellOrder, buyerAc
 			return err
 		}
 		supply.RetiredAmount = supplyRetired.String()
+
+		supplyTradable, err := math.NewDecFromString(supply.TradableAmount)
```

Comment:
After these two txns
```
{"entry_kind":"msg","height":24,"order":296,"operation":{"route":"/regen.ecocredit.marketplace.v1.MsgSell","name":"/regen.ecocredit.marketplace.v1.MsgSell","comment":"","ok":true,"msg":{"orders":[{"ask_price":{"amount":"24","denom":"stake"},"batch_denom":"C01-41900513-41900613-001","expiration":"4191-05-15T12:45:06Z","quantity":"33"}],"owner":"regen14c3j672kvw9l5uleh4x9uds2fre5vl7y60salu"}}}
{"entry_kind":"msg","height":24,"order":306,"operation":{"route":"/regen.ecocredit.marketplace.v1.MsgBuyDirect","name":"/regen.ecocredit.marketplace.v1.MsgBuyDirect","comment":"","ok":true,"msg":{"buyer":"regen1exnqjlk9g477w80vqvm7scda7l4fcjzfyvx5rf","orders":[{"bid_price":{"amount":"49","denom":"stake"},"quantity":"33","retirement_jurisdiction":"AQ","sell_order_id":"1"}]}}}
```
Invariants are breaking with the following error

```
panic: invariant broken: ecocredit: batch-supply invariant
        tradable supply is incorrect for 1 credit batch, expected 100, got 67

```

Reply:
I can have a fix for this up today

Reply:
Can you open an issue for this @technicallyty for tracking? :pray:

Reply:
https://github.com/regen-network/regen-ledger/issues/1067